crates/shim/src/mount.rs

```diff
@@ -239,49 +238,52 @@ fn options_size(options: &[String]) -> usize {
     options.iter().fold(0, |sum, x| sum + x.len())
 }
 
-fn longest_common_prefix(dirs: &[String]) -> Option<String> {
+fn longest_common_prefix(dirs: &[String]) -> &str {
     if dirs.is_empty() {
-        return None;
-    }
-    if dirs.len() == 1 {
-        if dirs[0].is_empty() {
-            return None;
-        }
-        return Some(dirs[0].to_string());
+        return "";
     }
 
-    let min = dirs.iter().min().unwrap();
-    let max = dirs.iter().max().unwrap();
-    let min_chars = min.chars().collect::<Vec<char>>();
-    let max_chars = max.chars().collect::<Vec<char>>();
-    let mut i = 0;
-    while i < min_chars.len() && i < max_chars.len() {
-        if min_chars[i] != max_chars[i] {
-            if i == 0 {
-                return None;
+    let first_dir = &dirs[0];
+
+    for (i, byte) in first_dir.as_bytes().iter().enumerate() {
+        for dir in dirs {
+            if dir.as_bytes().get(i) != Some(byte) {
+                let mut end = i;
+                // guaranteed not to underflow since is_char_boundary(0) is always true
+                while !first_dir.is_char_boundary(end) {
+                    end -= 1;
+                }
+
+                return &first_dir[0..end];
             }
-            return Some(min[0..i].to_string());
         }
-        i += 1;
     }
 
-    if min.is_empty() {
-        None
-    } else {
-        Some(min.to_string())
-    }
+    first_dir
 }
 
 // NOTE: the snapshot id is based on digits.
-// in order to avoid to get snapshots/x, shoule be back to parent dir.
+// in order to avoid to get snapshots/x, should be back to parent dir.
 // however, there is assumption that the common dir is ${root}/io.containerd.v1.overlayfs/snapshots.
 #[cfg(target_os = "linux")]
 fn trim_flawed_dir(s: &str) -> String {
-    let r = Regex::new(r"((/[^/]+)+/)([^/]*)").unwrap();
-    r.replace(s, "$1").to_string()
+    match s.ends_with('/') {
+        true => s.to_owned(),
+        false => {
+            // Iterate in reverse order to find the last '/', then return string in correct order
+            s.chars()
+                .rev()
+                .skip_while(|x| *x != '/')
```

Comment:
Would [`rfind`](https://doc.rust-lang.org/std/string/struct.String.html#method.rfind) be a better fit for this?

Reply:
#TIL - yes, that looks much better, thank you! Updated in https://github.com/containerd/rust-extensions/pull/201/commits/9e2c7a7113a1858b8e273fe94d1abd35a09b8ea9